### PR TITLE
Improve MVC Rest exception handling and returned content type.

### PR DESCRIPTION
**Pull request by fernandor777**
This PR improves RestException handling moving to central Controller Advise when can (since some controller need to have it locally by a test) and setting the right content type before returning ResponseEntity to Spring MVC layer.

Also added some full MVC integration test classes for testing exception handler.

### CODE REVIEW

**Comment by aaime** [comment]
Checkstyle failed with the following:

```
[INFO] --- maven-checkstyle-plugin:3.0.0:check (default) @ gwc-rest ---
[WARNING] File encoding has not been set, using platform encoding UTF-8, i.e. build is platform dependent!
[INFO] There is 1 error reported by Checkstyle 6.18 with /home/travis/build/GeoWebCache/geowebcache/geowebcache/checkstyle.xml ruleset.
[ERROR] src/main/java/org/geowebcache/rest/controller/RestExceptionHandler.java:[1] (header) RegexpHeader: Line does not match expected header line of '^/\*\*$'.
[INFO]
```

Indeed the file has a license header valid for GeoServer, but not matching the GWC ones (the license is different, for example).

File: geowebcache/rest/src/main/java/org/geowebcache/rest/service/SeedService.java [hunk 1]
```diff
@@ -107,15 +109,22 @@ public SeedService(ApplicationContextProvider appCtx) {
...
+                    headers.setContentType(MediaType.TEXT_PLAIN);
+                    return new ResponseEntity<String>(
+                            e.getMessage(), headers, HttpStatus.BAD_REQUEST);
                 }
                 list = seeder.getStatusList(layer);
             }
             obj = new JSONObject(xs.toXML(list));
-            return new ResponseEntity<>(obj.toString(), HttpStatus.OK);
+            HttpHeaders headers = new HttpHeaders();
+            headers.setContentType(MediaType.APPLICATION_JSON);
```
[the remark this answers is not in the document]
Comment (fernandor777) [reply geowebcache/rest/src/main/java/org/geowebcache/rest/service/SeedService.java]: Indeed, added explicit content type to other JSON and plain text response enttities on SeedService.

**Comment by fernandor777** [comment]
Changes done, fixed header on RestExceptionHandler class, thanks for your review.

**Comment by aaime** [comment]
Merged thank you. Can you do backports?

**Pull request merged**